website_multi_company/models/website.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class Website(models.Model):
+    _inherit = 'website'
+
+    @api.multi
+    @api.constrains('user_id')
+    def _check_user_id(self):
+        """Do not allow same public user on two websites.s"""
+        for record in self:
+            existing = self.search([('user_id', '=', record.user_id.id)])
+            if len(existing) > 1:
```

Comment:
`if existing:` is enough

Reply:
We're going for a dup check here, so I'm checking for >1

Reply:
OK, as the check is done also on write, you can have in the search the current record.